csbuild/tools/project_generators/visual_studio/internal/__init__.py

```diff
@@ -93,11 +97,15 @@ class Version(object):
 ALL_FILE_EXTENSIONS = CPP_SOURCE_FILE_EXTENSIONS \
 	| CPP_HEADER_FILE_EXTENSIONS \
 	| OBJC_SOURCE_FILE_EXTENSIONS \
+	| HLSL_SOURCE_FILE_EXTENSIONS \
+	| HLSL_HEADER_FILE_EXTENSIONS \
 	| ASM_FILE_EXTENSIONS \
+	| PYTHON_FILE_EXTENSIONS \
+	| BATCH_FILE_EXTENSIONS \
 	| MISC_FILE_EXTENSIONS
 
 # Switch for toggling the project folders separating files by their extensions.
-ENABLE_FILE_TYPE_FOLDERS = True
+ENABLE_FILE_TYPE_FOLDERS = False
```

Comment:
Why this change?

Reply:
Because while it serves a reasonable purpose for some, most projects disable it.  Made more sense to me to disable it by default.

Reply:
Ideally, this would be set through a command line argument, but we don't have the ability to define custom arguments yet.